libraries/Microsoft.Bot.Builder.Ai/LocaleConverterMiddleware.cs

```diff
@@ -27,12 +28,12 @@ public class LocaleConverterMiddleware : IMiddleware
         /// <param name="localeConverter">An ILocaleConverter instance</param>
         public LocaleConverterMiddleware(Func<ITurnContext, string> getUserLocale, Func<ITurnContext, Task<bool>> checkUserLocaleChanged, string toLocale, ILocaleConverter localeConverter)
         {
-            this.localeConverter = localeConverter ?? throw new ArgumentNullException(nameof(localeConverter));
+            this._localeConverter = localeConverter ?? throw new ArgumentNullException(nameof(localeConverter));
             if (string.IsNullOrEmpty(toLocale) || !localeConverter.IsLocaleAvailable(toLocale))
```

Comment:
I would break this "if" statement apart, if only to throw DIFFERENT exceptions. The toLocale being unset is an argument null exception. 

The local being unavailable seems like it should be a different exception, or at least a different error message ("The locale 'foo-za'  is unavailable".)

Reply:
done